src/datasets/utils/metadata.py

```diff
@@ -252,6 +252,8 @@ def to_dataset_card_data(self, dataset_card_data: DatasetCardData) -> None:
 
     def get_default_config_name(self) -> Optional[str]:
         default_config_name = None
+        if len(self) == 1:
+            return next(iter(self))
         for config_name, metadata_config in self.items():
             if config_name == "default" or metadata_config.get("default"):
```

Comment:
What about this refactoring to avoid having 2 returns?
```suggestion
        for config_name, metadata_config in self.items():
            if len(self) == 1 or config_name == "default" or metadata_config.get("default"):
```